hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/client/ReplicationConfigValidator.java

```diff
@@ -50,10 +50,11 @@ public void init() {
 
   public ReplicationConfig validate(ReplicationConfig replicationConfig) {
     if (validationRegexp != null) {
-      if (!validationRegexp.matcher(replicationConfig.toString()).matches()) {
+      if (!validationRegexp.matcher(
```

Comment:
The change looks good. A minor observation: We accept a regular expression as a configuration to limit the replication configurations, it would be useful to validate the regular expression? If we only support a subset of replication configurations would it be better to not validate via string comparison as we already have `ReplicationConfiguration` which has the full fidelity of the configuration already parsed into its constituents, it would be better to validate semantically vs via regular expressions.